store/mongo/datastore_mongo.go

```diff
@@ -558,6 +612,26 @@ func (db *DataStoreMongo) UpsertRemoveDeviceAttributes(
 		Database(mstore.DbFromContext(ctx, DbName)).
 		Collection(DbDevicesColl)
 
+	var filter interface{}
+	if etag != nil && *etag != "" {
+		filter = bson.M{
+			"_id":     id,
+			etagField: bson.M{"$exists": true, "$eq": *etag},
```

Comment:
Again, no need to use $exists and $eq together

Reply:
Makes sense. Thanks!